src/fides/api/ops/api/v1/endpoints/storage_endpoints.py

```diff
@@ -331,6 +336,90 @@ def get_active_default_config(
     return storage_config
 
 
+# this needs to come before other `/default/{storage_type}` routes so that `/status`
+# isn't picked up as a path param
+@router.get(
+    STORAGE_STATUS,
+    dependencies=[Security(verify_oauth_client, scopes=[STORAGE_READ])],
+    response_model=StorageConfigStatusMessage,
+    responses={
+        HTTP_200_OK: {
+            "content": {
+                "application/json": {
+                    "example": {
+                        "config_status": "configured",
+                        "detail": "Active default storage location of type s3 is fully configured",
+                    }
+                }
+            }
+        }
+    },
+)
+def get_storage_status(
+    *, db: Session = Depends(deps.get_db)
+) -> StorageConfigStatusMessage:
+    """
+    Determines the status of the active default storage config.
+
+    In order to be considered fully configured, the active default storage
+    config MUST be an s3 storage config.
+    """
+    logger.info("Determining active default storage config status")
+
+    # confirm an active default storage config is present
+    storage_config = get_active_default_storage_config(db)
+    if not storage_config:
+        return StorageConfigStatusMessage(
+            config_status=StorageConfigStatus.not_configured,
+            detail="No active default storage configuration found",
+        )
+
+    # at this point, we only treat the active default storage as fully configured
+    # if it is an s3 storage location. eventually we may need to adjust this when
+    # we support other object stores/storage destinations as valid for production.
+    if storage_config.type != StorageType.s3:
+        return StorageConfigStatusMessage(
+            config_status=StorageConfigStatus.not_configured,
+            detail="Active default storage configuration is not s3",
+        )
+
+    try:
+        details = storage_config.details
+        StorageDestinationBase.validate_details(details, storage_config.type.value)  # type: ignore
+    except Exception:
+        return StorageConfigStatusMessage(
+            config_status=StorageConfigStatus.not_configured,
+            detail=f"Invalid or unpopulated details on {storage_config.type.value} storage configuration",  # type: ignore
+        )
+
+    # currently, secrets are only required for s3 storage with auth method `SECRET_KEYS`
+    secrets = storage_config.secrets
+    if (
```

Comment:
same here, can we make this logic more scalable to handle additional supported types in the future?

Reply:
hmm, yeah, let me see. for this one, abstracting it out may make it more confusing and not worth the effort at this point, but i'll see what i can come up with.

Reply:
updated in https://github.com/ethyca/fides/commit/f5b36f74bfe0251f1725f6c9b7cc1570e82b2f44, resolving for now but we can create a follow-up if it needs any adjustment